share/completions/diskutil.fish

```diff
@@ -1,5 +1,27 @@
 # completion for diskutil (macOS)
 
+#############
+# Functions #
+#############
+
+function __fish_diskutil_needs_command
+  set cmd (commandline -opc)
+  if test (count $cmd) -eq 1 -a $cmd[1] = 'diskutil'
```

Comment:
This pattern is surprisingly common, yet it is _wrong_.

For one, checking that the first token is the name of the command (`$cmd[1] = diskutil`) isn't necessary and breaks aliases. This also doesn't work with boolean commands (`and diskutil` - but that also breaks `__fish_use_subcommand`).

Also, the double test conditions don't work like you think. If `$cmd` empty, then this will be expanded _before_ the `count` is run, which means it'll expand to `test (count) -eq 1 -a = diskutil` - which will cause an error about a missing operand. This won't happen here since the diskutil completions have been triggered, so there _must_ be something in the commandline, but it's a misconception.

`test` is not part of the language - it's simply a command that happens to be a builtin for performance reasons.

Reply:
Anyway, please remove these two functions again.